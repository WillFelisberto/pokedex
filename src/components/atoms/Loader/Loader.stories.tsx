import type { Meta, StoryObj } from '@storybook/react';

import { Loader } from './Loader';

const meta: Meta<typeof Loader> = {
  title: 'Atoms/Loader',
  component: Loader,
  tags: ['autodocs'],
  argTypes: {
    fullScreen: {
      control: 'boolean',
      description:
        'Defines whether the loader covers the entire screen or just the spinning Pok√©ball',
      defaultValue: false
    }
  }
};

export default meta;

type Story = StoryObj<typeof Loader>;

export const Default: Story = {
  args: {
    fullScreen: false
  }
};

export const FullScreen: Story = {
  args: {
    fullScreen: true
  }
};
