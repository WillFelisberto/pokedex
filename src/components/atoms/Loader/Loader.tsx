import Image from 'next/image';

interface LoaderProps {
  fullScreen?: boolean;
}

export const Loader = ({ fullScreen = true }: LoaderProps) => (
  <div
    className={`flex justify-center items-center ${
      fullScreen ? 'fixed top-0 left-0 w-screen h-screen bg-red-600 z-50' : ''
    }`}
  >
    <div className="w-24 h-24">
      {/* Pok√©bola girando */}
      <Image
        src="/pokeball.svg"
        width={100}
        height={100}
        alt="Loading..."
        className="w-full h-full animate-spin"
      />
    </div>
  </div>
);
