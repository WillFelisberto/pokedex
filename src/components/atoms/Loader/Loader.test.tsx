import { render, screen } from '@/tests/test-utils';

import { Loader } from './Loader';

describe('<Loader />', () => {
  it('should render the component', () => {
    const { container } = render(<Loader />);

    expect(screen.getByAltText('Loading...')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should render the full screen loader', () => {
    const { container } = render(<Loader />);

    expect(screen.getByAltText('Loading...')).toBeInTheDocument();
    expect(container.firstChild).toHaveClass('fixed');
    expect(container.firstChild).toHaveClass('w-screen h-screen bg-red-600');
  });

  it('should render only the spinning PokÃ©ball when fullScreen is false', () => {
    const { container } = render(<Loader fullScreen={false} />);

    expect(screen.getByAltText('Loading...')).toBeInTheDocument();
    expect(container.firstChild).not.toHaveClass('fixed');
    expect(container.firstChild).not.toHaveClass('bg-red-600');
  });
});
