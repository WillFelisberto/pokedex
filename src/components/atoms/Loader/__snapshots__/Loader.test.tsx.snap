// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should render the component 1`] = `
<div>
  <div
    class="flex justify-center items-center fixed top-0 left-0 w-screen h-screen bg-red-600 z-50"
  >
    <div
      class="w-24 h-24"
    >
      <img
        alt="Loading..."
        class="w-full h-full animate-spin"
        height="100"
        src="/pokeball.svg"
        width="100"
      />
    </div>
  </div>
</div>
`;
